### PR TITLE
RMET-594 Barcode Plugin ::: Android implementation

### PR DESCRIPTION
**Pull request by CarlsCorrea**
## Description

First implementation of the Barcode Plugin for Android using ZXing library with custom views

## Context

Closes: https://outsystemsrd.atlassian.net/browse/RMET-594

## Type of changes

- [ ] Fix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [X] Android
- [ ] iOS
- [ ] JavaScript

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Code follows code style of this project
- [ ] CHANGELOG.md file is correctly updated
- [x] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

**Review by andregrillo — APPROVED** [review]

File: plugin.xml [hunk 1]
```diff
@@ -8,13 +8,44 @@
   </js-module>
   
   <platform name="android">
+
+    <config-file target="AndroidManifest.xml" parent="application">        
+    <activity android:name=".CustomScannerActivity"
```
Comment (vmfo22) [inline plugin.xml:13]: Don't you need the full package name here?

File: src/android/com/outsystems/plugins/barcodescanner/OSBarcodeScanner.java [hunk 2]
```diff
@@ -1,30 +1,67 @@
...
+        }
+
+        IntentResult result = IntentIntegrator.parseActivityResult(resultCode, data);
+
+        if(result.getContents() == null) {
+            Intent originalIntent = result.getOriginalIntent();
+            if (originalIntent == null) {
+                _callbackContext.success("Cancelled");
+            } else if(originalIntent.hasExtra(Intents.Scan.MISSING_CAMERA_PERMISSION)) {
+                _callbackContext.success("Cancelled due to missing camera permission");
```
Comment (vmfo22) [inline src/android/com/outsystems/plugins/barcodescanner/OSBarcodeScanner.java:60]: Same question as above

**Review by vmfo22 — COMMENTED** [review]
Great Work @CarlsCorrea!
Check my questions

File: plugin.xml (hunk 1 above)
Comment (CarlsCorrea) [reply plugin.xml:13]: For now it seems to be working. I will check if it is necessary to change or not with mabs in the future.

File: src/android/com/outsystems/plugins/barcodescanner/OSBarcodeScanner.java [hunk 3]
```diff
@@ -1,30 +1,67 @@
...
+            super.onActivityResult(requestCode, resultCode, data);
+            return;
+        }
+
+        IntentResult result = IntentIntegrator.parseActivityResult(resultCode, data);
+
+        if(result.getContents() == null) {
+            Intent originalIntent = result.getOriginalIntent();
+            if (originalIntent == null) {
+                _callbackContext.success("Cancelled");
```
[the remark this answers is not in the document]
Comment (CarlsCorrea) [reply src/android/com/outsystems/plugins/barcodescanner/OSBarcodeScanner.java:58]: fixed

**Review by vmfo22 — APPROVED** [review]

**Pull request merged**